File: rb/features.rb
# Bonequest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module BonequestFeatures
  def self.make_feature(name)
    case name
    when "base"
      BonequestBaseFeature.new
    when "ratelimit"
      BonequestRatelimitFeature.new
    when "retry"
      BonequestRetryFeature.new
    when "test"
      BonequestTestFeature.new
    when "timeout"
      BonequestTimeoutFeature.new
    else
      BonequestBaseFeature.new
    end
  end
end

File: py/features.py
# Bonequest SDK feature factory

from feature.base_feature import BonequestBaseFeature
from feature.test_feature import BonequestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BonequestBaseFeature(),
        "test": lambda: BonequestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
